util/s3/s3Helper_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2019 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package s3
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestGetS3UploadBucket(t *testing.T) {
+	tests := []struct {
+		name           string
+		getDefault     bool
+		wantBucketName string
+	}{
+		{name: "test1", wantBucketName: "test-bucket"},
+		{name: "test2", wantBucketName: "anotherbucket"},
+		{name: "test3", wantBucketName: ""},
+		{name: "test4", getDefault: true, wantBucketName: "algorand-uploads"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.getDefault {
+				os.Unsetenv("S3_UPLOAD_BUCKET")
+			} else {
+				os.Setenv("S3_UPLOAD_BUCKET", tt.wantBucketName)
+			}
+			if gotBucketName := GetS3UploadBucket(); gotBucketName != tt.wantBucketName {
+				t.Errorf("GetS3UploadBucket() = %v, want %v", gotBucketName, tt.wantBucketName)
+			}
+		})
+	}
+}
+
+func TestGetS3ReleaseBucket(t *testing.T) {
+	tests := []struct {
+		name           string
+		getDefault     bool
+		wantBucketName string
+	}{
+		{name: "test1", wantBucketName: "test-bucket"},
+		{name: "test2", wantBucketName: "anotherbucket"},
+		{name: "test3", wantBucketName: ""},
+		{name: "test4", getDefault: true, wantBucketName: "algorand-releases"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.getDefault {
+				os.Unsetenv("S3_RELEASE_BUCKET")
+			} else {
+				os.Setenv("S3_RELEASE_BUCKET", tt.wantBucketName)
+			}
+			if gotBucketName := GetS3ReleaseBucket(); gotBucketName != tt.wantBucketName {
+				t.Errorf("GetS3ReleaseBucket() = %v, want %v", gotBucketName, tt.wantBucketName)
+			}
+		})
+	}
+}
+
+func Test_getS3Region(t *testing.T) {
+	tests := []struct {
+		name       string
+		getDefault bool
+		wantRegion string
+	}{
+		{name: "test1", wantRegion: "us-east1"},
+		{name: "test2", wantRegion: "us-west2"},
+		{name: "test3", wantRegion: ""},
+		{name: "test3", getDefault: true, wantRegion: "us-east-1"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.getDefault {
+				os.Unsetenv("S3_REGION")
+			} else {
+				os.Setenv("S3_REGION", tt.wantRegion)
+			}
+			if gotRegion := getS3Region(); gotRegion != tt.wantRegion {
+				t.Errorf("getS3Region() = %v, want %v", gotRegion, tt.wantRegion)
+			}
+		})
+	}
+}
+
```

Comment:
Is it possible to check the session credentials in `gotHelper.session`? Would be nice to check directly that the anonymous credentials work. I guess it's implicitly tested in test5/test9 - maybe give those tests more specific names?